.teamcity/src/main/kotlin/configurations/GradleBuildConfigurationDefaults.kt

```diff
@@ -42,13 +42,23 @@ val m2CleanScriptWindows = """
     )
 """.trimIndent()
 
-val cleanAndroidUserHomeScriptUnixLike = """
-    rm -rf %teamcity.agent.jvm.user.home%/.android
+val checkCleanAndroidUserHomeScriptUnixLike = """
+    ANDROID_USER_HOME=%teamcity.agent.jvm.user.home%/.android
+    if [ -e ${'$'}ANDROID_USER_HOME ] ; then
+        tree ${'$'}ANDROID_USER_HOME
+        rm -rf ${'$'}ANDROID_USER_HOME
+        echo "${'$'}ANDROID_USER_HOME was polluted during the build"
+        exit 1
+    else
+        echo "${'$'}ANDROID_USER_HOME does not exist"
+    fi
 """.trimIndent()
 
-val cleanAndroidUserHomeScriptWindows = """
+val checkCleanAndroidUserHomeScriptWindows = """
     IF exist %teamcity.agent.jvm.user.home%\.android (
```

Comment:
I suppose the GE builds using Android builds will cause some problems here - let's see. And our builds from other branches.

Reply:
Let's do this in 2 steps: first, isolate the android user home to build directory, then wait for ~week to all branches picking up this change. I can ask GE team do this as well.